File: rb/features.rb
# PrankShow SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module PrankShowFeatures
  def self.make_feature(name)
    case name
    when "base"
      PrankShowBaseFeature.new
    when "ratelimit"
      PrankShowRatelimitFeature.new
    when "retry"
      PrankShowRetryFeature.new
    when "test"
      PrankShowTestFeature.new
    when "timeout"
      PrankShowTimeoutFeature.new
    else
      PrankShowBaseFeature.new
    end
  end
end

File: py/features.py
# PrankShow SDK feature factory

from feature.base_feature import PrankShowBaseFeature
from feature.test_feature import PrankShowTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PrankShowBaseFeature(),
        "test": lambda: PrankShowTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
